src/str_extract.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import os, sys, re
+import argparse
+
+def strextract(dir, suffix=None, path=False, all=False):  
+    # Expression régulière pour trouver des chaînes littérales entre guillemets simples ou doubles
+    # compiled_re = re.compile(r"^.*'.*'.*"'"'r".*"'"'r"$", re.IGNORECASE)
+    regex_chain = re.compile(r'(["\'])(.*?)\1', re.IGNORECASE)
```

Comment:
Excellent

Reply:
Thanks, but I've copied a code without understanding it, showing it to you, you've suggested the whole class to use this "re.compile", then to me to initiate a variable and compile the regex right after entering the function.

Reply:
At least you did it properly